### PR TITLE
[ci][auto/5] load flaky test from both yaml and DB

### PR DESCRIPTION
**Pull request by can-anyscale**
Allow rayci to load the list of flaky tests from both yaml and DB

Test:
- CI

### CODE REVIEW

File: ci/ray_ci/test_tester.py [hunk 1]
```diff
@@ -144,11 +159,24 @@ def test_get_all_test_query() -> None:
...
 
-    with TemporaryDirectory() as tmp:
+    with TemporaryDirectory() as tmp, mock.patch(
+        "ray_release.test.Test.gen_from_s3",
+        return_value=[
+            _stub_test({"name": "windows://t1_s3", "team": "core"}),
+            _stub_test({"name": "windows://t2_s3", "team": "ci"}),
+            _stub_test({"name": "linux://t3_s3", "team": "core"}),
+        ],
+    ):
```
Comment (aslonnie) [inline ci/ray_ci/test_tester.py:169]: nit: can we rename `_TEST_YAML` into `_FAKE_CORE_TEST_YAML`, or something that indicates it is the yaml file for core team?

File: ci/ray_ci/tester.py [hunk 2]
```diff
@@ -352,7 +353,18 @@ def _get_flaky_test_targets(
         yaml_dir = os.path.join(bazel_workspace_dir, "ci/ray_ci")
 
     with open(f"{yaml_dir}/{team}.tests.yml", "rb") as f:
+        # load flaky tests from yaml
         all_flaky_tests = yaml.safe_load(f)["flaky_tests"]
+        # load flaky tests from DB
+        all_flaky_tests.extend(
```
Comment (aslonnie) [inline ci/ray_ci/tester.py:359]: do we want to deduplicate (and sort)? what if both the yaml and the db contains the same test?

and could you add a test for that case?

File: ci/ray_ci/test_tester.py [hunk 3]
```diff
@@ -144,11 +159,24 @@ def test_get_all_test_query() -> None:
 def test_get_flaky_test_targets() -> None:
     _TEST_YAML = "flaky_tests: [windows://t1, //t2]"
 
-    with TemporaryDirectory() as tmp:
+    with TemporaryDirectory() as tmp, mock.patch(
+        "ray_release.test.Test.gen_from_s3",
```
Comment (aslonnie) [inline ci/ray_ci/test_tester.py:163]: could you add some test cases when one (or both) of the yaml / db is an empty list?

**Comment by can-anyscale** [comment]
sorted, de-duplicate + more tests

File: ci/ray_ci/tester.py [hunk 4]
```diff
@@ -352,7 +353,19 @@ def _get_flaky_test_targets(
...
+        all_flaky_tests.update(
+            [
+                # remove "linux:" prefix for linux tests to be consistent with the
+                # interface supported in the yaml file
+                test.get_name().lstrip("linux:")
+                for test in Test.gen_from_s3(prefix=f"{operating_system}:")
+                if test.get_oncall() == team and test.get_state() == TestState.FLAKY
+            ]
+        )
+        all_flaky_tests = sorted(all_flaky_tests)
```
Comment (aslonnie) [inline ci/ray_ci/tester.py:368]: maybe do a rename? don't use the same var name for set and list?

**Review by aslonnie — APPROVED** [review]

**Comment by can-anyscale** [comment]
renamed some var

**Pull request merged**